src/zarr/core/array.py

```diff
@@ -218,6 +289,69 @@ async def create(
         exists_ok: bool = False,
         data: npt.ArrayLike | None = None,
     ) -> AsyncArray:
+        """
+        Method to create a new asynchronous array instance.
+
+        Parameters
+        ----------
+        store : StoreLike
+            The store where the array will be created.
+        shape : ShapeLike
+            The shape of the array.
+        dtype : npt.DTypeLike
+            The data type of the array.
+        zarr_format : ZarrFormat, optional
+            The Zarr format version (default is 3).
+        fill_value : Any, optional
+            The fill value of the array (default is None).
+        attributes : dict[str, JSON], optional
+            The attributes of the array (default is None).
+        chunk_shape : ChunkCoords, optional
+            The shape of the array's chunks (default is None).
+        chunk_key_encoding : ChunkKeyEncoding, optional
+            The chunk key encoding (default is None).
+        codecs : Iterable[Codec | dict[str, JSON]], optional
+            The codecs used to encode the data (default is None).
+        dimension_names : Iterable[str], optional
+            The names of the dimensions (default is None).
+        chunks : ShapeLike, optional
+            The shape of the array's chunks (default is None).
+            V2 only. V3 arrays should not have 'chunks' parameter.
+        dimension_separator : Literal[".", "/"], optional
+            The dimension separator (default is None).
+            V2 only. V3 arrays cannot have a dimension separator.
+        order : Literal["C", "F"], optional
+            The order of the array (default is None).
+            V2 only. V3 arrays should not have 'order' parameter.
+        filters : list[dict[str, JSON]], optional
+            The filters used to compress the data (default is None).
+            V2 only. V3 arrays should not have 'filters' parameter.
+        compressor : dict[str, JSON], optional
+            The compressor used to compress the data (default is None).
+            V2 only. V3 arrays should not have 'compressor' parameter.
+        exists_ok : bool, optional
+            Whether to raise an error if the store already exists (default is False).
+        data : npt.ArrayLike, optional
+            The data to be inserted into the array (default is None).
+
+        Returns
+        -------
+        AsyncArray
+            The created asynchronous array instance.
+
+        Examples
+        --------
+        >>> import zarr
+        >>> store = zarr.store.MemoryStore(mode='w')
+        >>> async_arr = await zarr.core.array.AsyncArray.create(
+        >>>     store=store,
+        >>>     shape=(100,100),
+        >>>     chunks=(10,10),
+        >>>     dtype='i4',
+        >>>     fill_value=0)
+        <AsyncArray memory://140349042942400 shape=(100, 100) dtype=int32>
```

Comment:
Usually a good idea to remove things that are platform / session dependent, like the memory address of the array:

```suggestion
        <AsyncArray memory://... shape=(100, 100) dtype=int32>
```

You can include a `# doctest: +ELLIPSIS` at the end of the line and it should be runnable through doctest, but I don't think we're running doctests on the project right now.

Reply:
okay, thanks! So not running doctests for now, but fine to have in there for the future? I added it in but can remove if needed